### PR TITLE
RUMM-189 add benchmark test on File I/O

### PR DESCRIPTION
**Pull request by xgouchet**
### What does this PR do?

Add benchmark test on File I/O

### Motivation

More test is always good :)

### Review checklist (to be filled by reviewers)

- [X] Feature or bugfix MUST have appropriate tests (unit, integration, e2e)
- [X] Make sure you discussed the feature or bugfix with the maintaining team in an Issue
- [X] Make sure each commit and the PR mention the Issue number (cf the [CONTRIBUTING](CONTRIBUTING.md) doc)

### CODE REVIEW

File: instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt [hunk 1]
```diff
@@ -0,0 +1,146 @@
...
+
+        println("Sleeping")
+        Thread.sleep(5500L)
+        println("Slept")
+
+        benchmark.measureRepeated {
+            println("Reading")
+            val batch = testedReader.invokeMethod("readNextBatch")
+            runWithTimingDisabled {
+                println("Read $batch")
```
Comment (ncreated) [inline instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt]: Do we need this `println()s`? I think they might add additional time for writing to stdo.

File: instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt [hunk 2]
```diff
@@ -0,0 +1,146 @@
...
+                checkNotNull(batch)
+                val id = batch.invokeMethod("getId") as String
+                testedReader.invokeMethod("releaseBatch", id)
```
Comment (mariusc83) [inline instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt]: maybe you will also want to benchmark this part ? Maybe keep it outside of the closure ?

File: tools/unit/src/main/kotlin/com/datadog/tools/unit/ReflectUtils.kt [hunk 3]
```diff
@@ -74,28 +94,116 @@ inline fun <reified T> Any.getFieldValue(fieldName: String): T {
...
         throw e.cause ?: e
+    } finally {
+        method.isAccessible = wasAccessible
     }
+
+    return output
+}
+
+@Suppress("TooGenericExceptionCaught", "SwallowedException", "SpreadOperator")
+private fun <T : Any> T.getDeclaredMethodRecursively(
```
Comment (mariusc83) [inline tools/unit/src/main/kotlin/com/datadog/tools/unit/ReflectUtils.kt]: nice ;)

File: instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt (hunk 1 above)
Comment (xgouchet) [reply instrumented/benchmark/src/androidTest/kotlin/com/datadog/android/sdk/benchmark/log/LogIOBenchmark.kt]: Indeed, I was trying to debug something, I'll remove them :)

**Pull request merged**